### PR TITLE
git-remote-hg: fix; refactor

### PR DESCRIPTION
**Pull request by alyssais**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/version-management/git-and-tools/git-remote-hg/default.nix [hunk 1]
```diff
@@ -1,36 +1,38 @@
-{ stdenv, lib, fetchFromGitHub, mercurial, makeWrapper
+{ stdenv, lib, fetchFromGitHub, python3Packages, makeWrapper
 , asciidoc, xmlto, docbook_xsl, docbook_xml_dtd_45, libxslt, libxml2
 }:
 
 stdenv.mkDerivation rec {
```
Comment (worldofpeace) [inline pkgs/applications/version-management/git-and-tools/git-remote-hg/default.nix:5]: Can't we use buildPythonApplication? This brings in wrapPython and does that postFixup thingy etc.

File: pkgs/applications/version-management/git-and-tools/git-remote-hg/default.nix [hunk 2]
```diff
@@ -1,36 +1,33 @@
...
 
-  buildInputs = [ mercurial.python mercurial makeWrapper
+  nativeBuildInputs = [
     asciidoc xmlto docbook_xsl docbook_xml_dtd_45 libxslt libxml2
   ];
 
-  doCheck = false;
-
-  installFlags = [ "HOME=\${out}" "install-doc" ];
+  pythonPath = with python3Packages; [ mercurial ];
```
Comment (jonringer) [inline pkgs/applications/version-management/git-and-tools/git-remote-hg/default.nix:20]: now that it's a pythonApplication
```suggestion
  propagatedBuilInputs = with python3Packages; [ mercurial ];
```

but it doesn't really matter

**Review by jonringer — APPROVED** [review]
diff LGTM

File: pkgs/applications/version-management/git-and-tools/git-remote-hg/default.nix (hunk 2 above)
Comment (alyssais) [reply pkgs/applications/version-management/git-and-tools/git-remote-hg/default.nix:20]: How come it needs to be propagated? This isn’t a library, so it should only be used through the executable wrapper, which includes Mercurial anyway, right?

Comment (jonringer) [reply pkgs/applications/version-management/git-and-tools/git-remote-hg/default.nix:20]: it's how python packaging works on nixpkgs https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonapplication-function, the transitive python package set is constructed off of propagated python packages.

`buildPythonApplication` will "end" this propagation to other packages which use `git-remote-hg`

**Review by worldofpeace — APPROVED** [review]

**Pull request merged**